File: mobile/src/types/workSchedule.ts
/** 영농 일정(Schedule) — WLS-001 */

export const SCHED_STATUS_PENDING = 'WS010100'
export const SCHED_STATUS_CONVERTED = 'WS010200'
export const SCHED_STATUS_CANCELLED = 'WS010300'

export type WorkScheduleItem = {
  farm_cd: string
  sched_id: string
  work_dt: string
  work_main_cd: string
  work_mid_cd: string
  work_loc_id?: string | null
  title?: string | null
  contents?: string | null
  sched_status_cd: string
  converted_work_id?: string | null
  google_event_id?: string | null
  sync_status?: string
  last_synced_at?: string | null
}

export type WorkScheduleListResponse = {
  success: boolean
  data: WorkScheduleItem[]
}

export type WorkScheduleCreatePayload = {
  work_dt: string
  work_mid_cd: string
  work_loc_id?: string | null
  title?: string | null
  contents?: string | null
}

export type WorkScheduleCreateResponse = {
  success: boolean
  data: {
    sched_id: string
    sched_status_cd: string
  }
}

export type WorkScheduleConvertResponse = {
  success: boolean
  data: {
    sched_id: string
    work_id: string
    prefilled_data: {
      work_dt: string
      work_mid_cd: string
      work_loc_id?: string | null
      memo: string
    }
  }
}

File: server/schemas/work_schedule.py
# -*- coding: utf-8 -*-
"""영농 일정(Schedule) API 스키마 — WLS-001 Phase1."""

from __future__ import annotations

from pydantic import BaseModel, Field


class WorkScheduleItem(BaseModel):
    farm_cd: str
    sched_id: str
    work_dt: str
    work_main_cd: str = "WK01"
    work_mid_cd: str
    work_loc_id: str | None = None
    title: str | None = None
    contents: str | None = None
    sched_status_cd: str
    converted_work_id: str | None = None
    google_event_id: str | None = None
    sync_status: str = "PENDING"
    last_synced_at: str | None = None


class WorkScheduleListResponse(BaseModel):
    success: bool = True
    data: list[WorkScheduleItem] = Field(default_factory=list)


class WorkScheduleCreateRequest(BaseModel):
    work_dt: str
    work_mid_cd: str
    work_loc_id: str | None = None
    title: str | None = None
    contents: str | None = None


class WorkScheduleUpdateRequest(BaseModel):
    work_dt: str | None = None
    work_mid_cd: str | None = None
    work_loc_id: str | None = None
    title: str | None = None
    contents: str | None = None
    sched_status_cd: str | None = None


class WorkScheduleCreateResponse(BaseModel):
    success: bool = True
    data: dict = Field(default_factory=dict)


class WorkScheduleConvertPrefill(BaseModel):
    work_dt: str
    work_mid_cd: str
    work_loc_id: str | None = None
    memo: str = ""


class WorkScheduleConvertData(BaseModel):
    sched_id: str
    work_id: str
    prefilled_data: WorkScheduleConvertPrefill


class WorkScheduleConvertResponse(BaseModel):
    success: bool = True
    data: WorkScheduleConvertData


class WorkScheduleMessageResponse(BaseModel):
    success: bool = True
    message: str = ""
